src/modules/mim/daemonconfiguration.json

```diff
@@ -0,0 +1,143 @@
+                          "name": "unknown",
+                          "enumValue": 0
+                        },
+                        {
+                          "name": "running",
+                          "enumValue": 1
+                        },
+                        {
+                          "name": "failed",
+                          "enumValue": 2
+                        },
+                        {
+                          "name": "exited",
+                          "enumValue": 3
+                        },
+                        {
+                          "name": "dead",
+                          "enumValue": 4
+                        }
+                      ]
+                    }
+                  },
+                  {
+                    "name": "status",
+                    "schema": {
+                      "type": "enum",
+                      "valueSchema": "integer",
+                      "enumValues": [
+                        {
+                          "name": "unknown",
+                          "enumValue": 0
+                        },
+                        {
+                          "name": "enabled",
+                          "enumValue": 1
+                        },
+                        {
+                          "name": "enabledRuntime",
+                          "enumValue": 2
+                        },
+                        {
+                          "name": "disabled",
+                          "enumValue": 3
+                        },
+                        {
+                          "name": "static",
+                          "enumValue": 4
+                        },
+                        {
+                          "name": "indirect",
+                          "enumValue": 5
+                        }
+                      ]
+                    }
+                  }
+                ]
+              }
+            }
+          },
+          {
+            "name": "daemonCommands",
+            "type": "mimObject",
+            "desired": true,
+            "schema": {
+              "type": "array",
+              "elementSchema": {
+                "type": "object",
+                "fields": [
+                  {
+                    "name": "name",
+                    "schema": "string"
+                  },
+                  {
+                    "name": "state",
+                    "schema": {
+                      "type": "enum",
+                      "valueSchema": "integer",
+                      "enumValues": [
+                        {
+                          "name": "run",
+                          "enumValue": 0
+                        },
+                        {
+                          "name": "stop",
+                          "enumValue": 1
+                        }
+                      ]
+                    }
+                  },
+                  {
+                    "name": "status",
```

Comment:
Merge state and status in one?

enable, disable, start, stop, can be issued one at a time. 

Thus this can be a single enumeration for desired state of the deamon named by the name setting:

0 (none)
1 (enable)
2 (disable)
3 (start)
4 (stop)

Reply:
As discussed offline, we'll do for the desired object:

string name;
boolean enabled;
boolean started;